### PR TITLE
Add GraphQL documentation doc.

### PR DESCRIPTION
**Pull request by mihaimaruseac**
Cleanup GraphQL SDL, make the comments consistent.

Group trees into a story to give to users, making `GraphQL.md` doc more readable than the GraphQL SDL.

### CODE REVIEW

File: docs/GraphQL.md [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+ontology](https://github.com/guacsec/guac/blob/main/docs/ontology.md) and its
+[definition](https://github.com/guacsec/guac/blob/main/docs/ontology-definitions.md).
+
+Note: the GraphQL definitions are not yet stable, they might change in future
+code changes. To get an up to date view of the definitions, you can use one of
+the following:
+
+- Directly consult the source of truth represented by the [GraphQL SDL
+  schema](https://github.com/guacsec/guac/tree/main/pkg/assembler/graphql/schema).
+- Look at the generated Go documentation using `godoc` and analyzing the
```
Comment (lumjjb) [inline docs/GraphQL.md:13]: can we link an alternate link to the public godoc?

File: docs/GraphQL.md [hunk 2]
```diff
@@ -0,0 +1,755 @@
+# GUAC GraphQL definitions
```
Comment (lumjjb) [inline docs/GraphQL.md:1]: can we do a small section on what is GraphQL and link out to some graphQL getting started for those that are not familiar

**Review by lumjjb — APPROVED** [review]
minor nits, else LGTM! Thanks for the in-depth doc!

File: docs/GraphQL.md [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
+```
+
+Finally, the last evidence tree defined in GUAC is attaching SBOM information to
+packages and sources:
+
+```gql
+type HasSBOM {
+  id: ID!
+  "SBOM subject"
+  subject: PackageOrSource!
```
Comment (pxp928) [inline docs/GraphQL.md:596]: we will have to make a change here to replace source with artifact

**Review by pxp928 — APPROVED** [review]
LGTM! Very comprehensive!

File: docs/GraphQL.md (hunk 1 above)
Comment (jeffmendoza) [reply docs/GraphQL.md:13]: Looks like we need to do that after the next release.

**Review by jeffmendoza — APPROVED** [review]

File: docs/GraphQL.md (hunk 3 above)
Comment (mihaimaruseac) [reply docs/GraphQL.md:596]: @jeffmendoza will do this in subsequent PR

**Pull request merged**